sympy/sets/fancysets.py

```diff
@@ -497,30 +497,40 @@ def normalize_theta_set(theta):
 
     if theta.is_Interval:
         # one complete circle
-        if Abs(theta.args[0] - theta.args[1]) >= 2*S.Pi:
+        interval_len = theta.measure
+        if interval_len >= 2*S.Pi:
+            if interval_len == 2*S.Pi and theta.left_open and theta.right_open:
+                k = coeff(theta.start)
+                return Union(Interval(0, k*S.Pi, False, True),
+                        Interval(k*S.Pi, 2*S.Pi, True, True))
             return Interval(0, 2*S.Pi, False, True)
 
         new_theta = []
-        for val in [theta.args[0], theta.args[1]]:
-            k = coeff(val)
-            if (not k) and (k != S.Zero):
-                raise NotImplementedError('Normalizing theta without pi as'
-                                          'coefficient, is not Implemented.')
-            elif k == S.Zero:
-                if val == S.Zero:
-                    new_theta.append(S.Zero)
-                else:
-                    # when theta is n*pi
-                    new_theta.append(2*S.Pi)
-            else:
-                new_theta.append(k*S.Pi)
+        k_start, k_end = coeff(theta.start), coeff(theta.end)
+        flag_start, flag_end = theta.left_open, theta.right_open
+
+        if k_start is None or k_end is None:
+            raise NotImplementedError("Normalizing theta without pi as coefficient is "
+                                    "not yet implemented")
+        # for values like 2*n*pi
+        if k_start == S.Zero:
+            new_theta.append((S.Zero, flag_start))
+        # for values like k*pi, k != 2*n
+        else:
+            new_theta.append((k_start*S.Pi, flag_start))
 
-        # for negative theta
-        if new_theta[0] > new_theta[1]:
-            return Union(Interval(S(0), new_theta[1]),
-                         Interval(new_theta[0], 2*S.Pi, False, True))
+        # similarly for end point
+        if k_end == S.Zero:
+            new_theta.append((S.Zero, flag_end))
         else:
-            return Interval(*new_theta)
+            new_theta.append((k_end*S.Pi, flag_end))
+
+        if new_theta[0][0] > new_theta[1][0]:
+            return Union(Interval(S(0), new_theta[1][0], False, new_theta[1][1]),
```

Comment:
Perhaps it could be possible to use ``theta.right_open`` instead of ``new_theta[1][1]`` and similarly for the other flags. Then in would not be necessary to add the flags to ``new_theta``.